lib/zendesk_api/middleware/response/parse_iso_dates.rb

```diff
@@ -19,7 +19,9 @@ def parse_dates!(value)
           case value
           when Hash then value.each { |key,element| value[key] = parse_dates!(element) }
           when Array then value.each_with_index { |element, index| value[index] = parse_dates!(element) }
-          when /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m then Time.parse(value)
+          when String
+            return Time.parse(value) if value.valid_encoding? && value =~ /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m
+            value
```

Comment:
otherwise we pass invalid encoded string in ... sounds like this will come back to fail another day ...

how about cleaning it up: `value.encode(Encoding::UTF_8, value.encoding, :invalid => :replace, :undef => :replace)`